### ISSUE TITLE
router.push({ hash }) not triggering a re-render

### ISSUE THREAD

**Issue by flpwgr** [scottrippey/next-router-mock#84]
Hi there!

I was playing around with mock router and found this peculiar edge case, maybe I missing something.

```javascript

jest.mock('next/router', () => require('next-router-mock'));

const ExampleComponent = ({ href = '' }) => {
  const router = useRouter();
  return (
    <button onClick={() => router.push({ hash: `#${href}` })}>
      The current route is: "{router.asPath}"
    </button>
  );
};

describe('next-router-mock', () => {
  it('mocks the useRouter hook', () => {
    // Set the initial url:
    mockRouter.push('/initial-path');

    // Render the component:
    render(<ExampleComponent href="test" />);
    expect(screen.getByRole('button')).toHaveTextContent(
      'The current route is: "/initial-path"',
    );

    // Click the button:
    fireEvent.click(screen.getByRole('button'));

    // Ensure the router was updated:
    expect(mockRouter).toMatchObject({
      asPath: '/initial-path#test',
      pathname: '/initial-path',
    });

    // THIS WILL THROW
    expect(screen.getByRole('button')).toHaveTextContent(
      'The current route is: "/initial-path#test"',
    );
  });
});
```

The error:
```
Expected element to have text content:
  The current route is: "/initial-path#test"
Received:
  The current route is: "/initial-path"
````

Apparently when we update just the hash the asPath is not updating in the component (although in next it does)

Mac OS Ventura 13.4
next@12.3.4
jest@28.1.3
next-router-mock@0.9.5

**Comment by scottrippey** [comment]
Nice find!  Easy fix!  Published as `0.9.6`

**Comment by flpwgr** [comment]
Woew ! Really fast! I was about to push the fix and your PR was ready! 😂
